// @ionic-native/intercom/index.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Cordova, IonicNativePlugin, Plugin } from '@ionic-native/core';
import { Injectable } from '@angular/core';
/**
 * @name Intercom
 * @description
 * This is a plugin that allows your Ionic app to use Intercom for iOS and/or Intercom for Android.
 * Follow the offical documentation to setup this plugin correctly: https://developers.intercom.com/docs/cordova-phonegap-configuration
 *
 * @usage
 * ```typescript
 * import { Intercom } from '@ionic-native/intercom';
 *
 *
 * constructor(private intercom: Intercom) { }
 *
 * ...
 *
 * this.intercom.registerUnidentifiedUser();
 * ...
 * this.intercom.registerForPush();
 *
 * ```
 */
var Intercom = (function (_super) {
    __extends(Intercom, _super);
    function Intercom() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * Register a identified user
     * @param options {any} Options
     * @return {Promise<any>} Returns a promise
     */
    /**
       * Register a identified user
       * @param options {any} Options
       * @return {Promise<any>} Returns a promise
       */
    Intercom.prototype.registerIdentifiedUser = /**
       * Register a identified user
       * @param options {any} Options
       * @return {Promise<any>} Returns a promise
       */
    function (options) { return; };
    /**
     * Register a unidentified user
     * @param options {any} Options
     * @return {Promise<any>} Returns a promise
     */
    /**
       * Register a unidentified user
       * @param options {any} Options
       * @return {Promise<any>} Returns a promise
       */
    Intercom.prototype.registerUnidentifiedUser = /**
       * Register a unidentified user
       * @param options {any} Options
       * @return {Promise<any>} Returns a promise
       */
    function (options) { return; };
    /**
     * This resets the Intercom integration's cache of your user's identity and wipes the slate clean.
     * @return {Promise<any>} Returns a promise
     */
    /**
       * This resets the Intercom integration's cache of your user's identity and wipes the slate clean.
       * @return {Promise<any>} Returns a promise
       */
    Intercom.prototype.reset = /**
       * This resets the Intercom integration's cache of your user's identity and wipes the slate clean.
       * @return {Promise<any>} Returns a promise
       */
    function () { return; };
    /**
     *
     * @param secureHash {string}
     * @param secureData {any}
     * @return {Promise<any>} Returns a promise
     * @deprecated Use setUserHash instead as of Intercom Cordova 4.0.0 and higher https://github.com/intercom/intercom-cordova/blob/master/CHANGELOG.md#400-2017-08-29
     */
    /**
       *
       * @param secureHash {string}
       * @param secureData {any}
       * @return {Promise<any>} Returns a promise
       * @deprecated Use setUserHash instead as of Intercom Cordova 4.0.0 and higher https://github.com/intercom/intercom-cordova/blob/master/CHANGELOG.md#400-2017-08-29
       */
    Intercom.prototype.setSecureMode = /**
       *
       * @param secureHash {string}
       * @param secureData {any}
       * @return {Promise<any>} Returns a promise
       * @deprecated Use setUserHash instead as of Intercom Cordova 4.0.0 and higher https://github.com/intercom/intercom-cordova/blob/master/CHANGELOG.md#400-2017-08-29
       */
    function (secureHash, secureData) { return; };
    /**
     *
     * @param secureHash {string}
     * @return {Promise<any>} Returns a promise
     */
    /**
       *
       * @param secureHash {string}
       * @return {Promise<any>} Returns a promise
       */
    Intercom.prototype.setUserHash = /**
       *
       * @param secureHash {string}
       * @return {Promise<any>} Returns a promise
       */
    function (secureHash) { return; };
    /**
     *
     * @param attributes {any}
     * @return {Promise<any>} Returns a promise
     */
    /**
       *
       * @param attributes {any}
       * @return {Promise<any>} Returns a promise
       */
    Intercom.prototype.updateUser = /**
       *
       * @param attributes {any}
       * @return {Promise<any>} Returns a promise
       */
    function (attributes) { return; };
    /**
     *
     * @param eventName {string}
     * @param metaData {any}
     * @return {Promise<any>} Returns a promise
     */
    /**
       *
       * @param eventName {string}
       * @param metaData {any}
       * @return {Promise<any>} Returns a promise
       */
    Intercom.prototype.logEvent = /**
       *
       * @param eventName {string}
       * @param metaData {any}
       * @return {Promise<any>} Returns a promise
       */
    function (eventName, metaData) { return; };
    /**
     *
     * @return {Promise<any>} Returns a promise
     */
    /**
       *
       * @return {Promise<any>} Returns a promise
       */
    Intercom.prototype.displayMessenger = /**
       *
       * @return {Promise<any>} Returns a promise
       */
    function () { return; };
    /**
     *
     * @return {Promise<any>} Returns a promise
     */
    /**
       *
       * @return {Promise<any>} Returns a promise
       */
    Intercom.prototype.displayMessageComposer = /**
       *
       * @return {Promise<any>} Returns a promise
       */
    function () { return; };
    /**
     *
     * @param initialMessage {string}
     * @return {Promise<any>} Returns a promise
     */
    /**
       *
       * @param initialMessage {string}
       * @return {Promise<any>} Returns a promise
       */
    Intercom.prototype.displayMessageComposerWithInitialMessage = /**
       *
       * @param initialMessage {string}
       * @return {Promise<any>} Returns a promise
       */
    function (initialMessage) { return; };
    /**
     *
     * @return {Promise<any>} Returns a promise
     */
    /**
       *
       * @return {Promise<any>} Returns a promise
       */
    Intercom.prototype.displayConversationsList = /**
       *
       * @return {Promise<any>} Returns a promise
       */
    function () { return; };
    /**
     *
     * @return {Promise<any>} Returns a promise
     */
    /**
       *
       * @return {Promise<any>} Returns a promise
       */
    Intercom.prototype.unreadConversationCount = /**
       *
       * @return {Promise<any>} Returns a promise
       */
    function () { return; };
    /**
     *
     * @param visibility {string}
     * @return {Promise<any>} Returns a promise
     */
    /**
       *
       * @param visibility {string}
       * @return {Promise<any>} Returns a promise
       */
    Intercom.prototype.setLauncherVisibility = /**
       *
       * @param visibility {string}
       * @return {Promise<any>} Returns a promise
       */
    function (visibility) { return; };
    /**
     *
     * @param visibility {string}
     * @return {Promise<any>} Returns a promise
     */
    /**
       *
       * @param visibility {string}
       * @return {Promise<any>} Returns a promise
       */
    Intercom.prototype.setInAppMessageVisibility = /**
       *
       * @param visibility {string}
       * @return {Promise<any>} Returns a promise
       */
    function (visibility) { return; };
    /**
     *
     * @return {Promise<any>} Returns a promise
     */
    /**
       *
       * @return {Promise<any>} Returns a promise
       */
    Intercom.prototype.hideMessenger = /**
       *
       * @return {Promise<any>} Returns a promise
       */
    function () { return; };
    /**
     *
     * @return {Promise<any>} Returns a promise
     */
    /**
       *
       * @return {Promise<any>} Returns a promise
       */
    Intercom.prototype.registerForPush = /**
       *
       * @return {Promise<any>} Returns a promise
       */
    function () { return; };
    Intercom.decorators = [
        { type: Injectable },
    ];
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "registerIdentifiedUser", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "registerUnidentifiedUser", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "reset", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [String, Object]),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "setSecureMode", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [String]),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "setUserHash", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "updateUser", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [String, Object]),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "logEvent", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "displayMessenger", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "displayMessageComposer", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [String]),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "displayMessageComposerWithInitialMessage", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "displayConversationsList", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "unreadConversationCount", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [String]),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "setLauncherVisibility", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [String]),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "setInAppMessageVisibility", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "hideMessenger", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], Intercom.prototype, "registerForPush", null);
    /**
     * @name Intercom
     * @description
     * This is a plugin that allows your Ionic app to use Intercom for iOS and/or Intercom for Android.
     * Follow the offical documentation to setup this plugin correctly: https://developers.intercom.com/docs/cordova-phonegap-configuration
     *
     * @usage
     * ```typescript
     * import { Intercom } from '@ionic-native/intercom';
     *
     *
     * constructor(private intercom: Intercom) { }
     *
     * ...
     *
     * this.intercom.registerUnidentifiedUser();
     * ...
     * this.intercom.registerForPush();
     *
     * ```
     */
    Intercom = __decorate([
        Plugin({
            pluginName: 'Intercom',
            plugin: 'cordova-plugin-intercom',
            pluginRef: 'intercom',
            repo: 'https://github.com/intercom/intercom-cordova',
            platforms: ['Android', 'iOS'],
        })
    ], Intercom);
    return Intercom;
}(IonicNativePlugin));
export { Intercom };
